src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -779,7 +783,21 @@ def __call__(
         # Offload all models
         self.maybe_free_model_hooks()
 
+        workflow = None
+        if return_workflow:
+            signature = inspect.signature(self.__call__)
+            argument_names = [param.name for param in signature.parameters.values()]
+            call_arg_values = inspect.getargvalues(inspect.currentframe()).locals
+            workflow = populate_workflow_from_pipeline(
+                argument_names, call_arg_values, self.lora_info, self.config._name_or_path
+            )
```

Comment:
```suggestion
            signature = inspect.signature(self.__call__)
            argument_names = [param.name for param in signature.parameters.values()]
            call_arg_values = inspect.getargvalues(inspect.currentframe()).locals
            workflow = populate_workflow_from_pipeline(
                argument_names, call_arg_values, self.lora_info, self.config._name_or_path
            )
```
should we move this to `pipeline_utils`?

Reply:
We can but the problem is the scope. If you delegate this to another module, the scope gets disturbed and we no longer get access to the right values.